module/zfs/dmu_send.c

```diff
@@ -612,6 +612,7 @@ send_traverse_thread(void *arg)
 	data = kmem_zalloc(sizeof (*data), KM_SLEEP);
 	data->eos_marker = B_TRUE;
 	bqueue_enqueue(&st_arg->q, data, 1);
+	thread_exit();
```

Comment:
Under Linux `thread_exit()` must be the last line of the function because it will never return.  In this case that's harmless because we don't really need to call `spl_fstrans_unmark()` since the threads existing anyway, but it does make the following line unreachable.  So let's move it after `spl_fstrans_unmark()` here and in `receive_writer_thread()`.